src/core/qgsfieldformatter.h

```diff
@@ -0,0 +1,87 @@
+/***************************************************************************
+  qgsfieldformatter.h - QgsFieldFormatter
+
+ ---------------------
+ begin                : 2.12.2016
+ copyright            : (C) 2016 by Matthias Kuhn
+ email                : matthias@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSFIELDKIT_H
+#define QGSFIELDKIT_H
+
+#include <QString>
+#include <QVariant>
+
+class QgsVectorLayer;
+
+/**
+ * \ingroup core
+ * A field kit helps to handle and display values for a field.
+ *
+ * It allows for using a shared configuration with the editor widgets
+ * for representation of attribute values.
+ * Field kits normally have one single instance which is managed by the
+ * QgsFieldFormatterRegistry. Custom field kits should be registered there and
+ * field kits for use within code should normally be obtained from there.
+ *
+ * This is an abstract base class and will always need to be subclassed.
+ *
+ * @note added in QGIS 3.0
+ */
+class CORE_EXPORT QgsFieldFormatter
+{
+  public:
+    QgsFieldFormatter();
+
+    virtual ~QgsFieldFormatter();
+
+    /**
+     * Return a unique id for this field kit.
+     * This id will later be used to identify this field kit in the registry with QgsFieldFormatterRegistry::fieldKit().
+     *
+     * This id matches the id of a QgsEditorWidgetFactory.
+     */
+    virtual QString id() const = 0;
+
+    /**
+     * Create a pretty String representation of the value.
+     *
+     * @return By default the string representation of the provided value as implied by the field definition is returned.
+     *
```

Comment:
Also a straight copy-paste, but yeah, that would really benefit from some documentation.